internal/handlers/handler.go

```diff
@@ -45,7 +46,8 @@ func ReadBody(w http.ResponseWriter, r *http.Request) ([]byte, error) {
 }
 
 func (sh StorageHandlers) PingDB(w http.ResponseWriter, r *http.Request) {
-	err := sh.storage.Ping()
+	ctx := context.Background()
```

Comment:
здесь и далее во всех хендлерах:
```suggestion
	ctx := r.Context()
```
https://pkg.go.dev/net/http#Request.Context

Подробнее о контексте: 
https://habr.com/ru/company/nixys/blog/461723/
https://blog.ildarkarymov.ru/posts/context-guide/